spec/features/settings/dex_connector_ldap_feature_spec.rb

```diff
@@ -134,67 +134,51 @@
   describe "#new_save_disabled_by_change" do
     before do
       visit new_settings_dex_connector_ldap_path
-      have_button("Save", disabled: true)
+      expect(page).to have_button("Save", disabled: true)
 
       # Test that we can enable the button
       page.execute_script("$('#ldap_conn_save').removeProp('disabled')")
-      have_button("Save", disabled: false)
+      expect(page).to have_button("Save", disabled: false)
     end
 
     it "allows change of start_tls" do
-      # Is false by default - change to true
-      # Following line should work but it does not so using JS
-      # choose id: "dex_connector_ldap_start_tls_true"
-      page.execute_script("$('#dex_connector_ldap_start_tls_true').prop('checked', true)")
-      page.execute_script("$('#dex_connector_ldap_start_tls_true').trigger('change')")
+      find("#dex_connector_ldap_start_tls_true_label").click
     end
 
     it "allows change of bind_anon" do
-      # Is false by default - change to true
-      # Following line should work but it does not so using JS
-      # choose id: "dex_connector_ldap_bind_anon_true"
-      page.execute_script("$('#dex_connector_ldap_bind_anon_true').prop('checked', true)")
-      page.execute_script("$('#dex_connector_ldap_bind_anon_true').trigger('change')")
+      find("#dex_connector_ldap_bind_anon_true_label").click
     end
 
     it "allows change of host" do
       fill_in "Host", with: "AAA"
-      page.execute_script("$('#dex_connector_ldap_host').trigger('change')")
     end
 
     it "allows change of port" do
       fill_in "Port", with: "AAA"
-      page.execute_script("$('#dex_connector_ldap_port').trigger('change')")
     end
 
     it "allows change certificate" do
       attach_file "Certificate", admin_cert_file.path
-      page.execute_script("$('#dex_connector_ldap_certificate').trigger('change')")
     end
 
     it "allows change of bind dn" do
-      fill_in id: "dex_connector_ldap_bind_dn", with: "AAA"
-      page.execute_script("$('#dex_connector_ldap_bind_dn').trigger('change')")
+      fill_in "dex_connector_ldap_bind_dn", with: "AAA"
     end
 
     it "allows change of password" do
-      fill_in id: "dex_connector_ldap_bind_pw", with: "AAA"
-      page.execute_script("$('#dex_connector_ldap_bind_pw[required=\"required\"]')"\
-        ".trigger('change')")
+      fill_in "dex_connector_ldap_bind_pw", with: "AAA"
```

Comment:
I feel that it warrants a comment in the code here noting that the lookup by id does not work. It is not clear why that should fail.

Reply:
It's actually looking by id, I just omitted the `id` key. The `fill_in` action will only fail if the `dex_connector_ldap_bind_pw` element is not found. When using `fill_in`, the field can be found via its name, id or label text. I think we could have used only "Password" here and it would also work but I didn't realize that in time.